uaclient/actions.py

```diff
@@ -161,6 +162,28 @@ def attach_with_token(
 
     machine_token_file.write(new_machine_token)
 
+    current_series = system.get_release_info().series
+    only_series = (
+        machine_token_file.entitlements()
+        .get("support", {})
+        .get("entitlement", {})
+        .get("affordances", {})
+        .get("onlySeries", None)
+    )
+    if only_series:
+        allowed_release = CODENAME_TO_RELEASE.get(only_series, "")
+        if only_series != current_series:
+            machine_token_file.delete()
```

Comment:
We just wrote the machine token at this point. Could this flow be executed before the above `write` call?

Reply:
@dheyay you moved this chunk of code but now we don't need the `delete` call right?